clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -24,123 +24,380 @@ package common
+type waitForControlledPodsRunningMeasurement struct {
+	informer          cache.SharedInformer
+	kind              string
+	namespace         string
+	labelSelector     string
+	fieldSelector     string
+	operationTimeout  time.Duration
+	stopCh            chan struct{}
+	isRunning         bool
+	queue             workqueue.Interface
+	workerGroup       wait.Group
+	handlingGroup     wait.Group
+	lock              sync.Mutex
+	opResourceVersion uint64
+	checkerMap        map[string]*objectChecker
+}
 
-// Execute waits until all specified RCs have all pods running or until timeout happens.
-// RCs can be specified by field and/or label selectors.
+// Execute waits until all specified controlling objects have all pods running or until timeout happens.
+// Controlling objects can be specified by field and/or label selectors.
 // If namespace is not passed by parameter, all-namespace scope is assumed.
-func (*waitForControlledPodsRunningMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+// "Start" action starts observation of the controlling objects, while "gather" waits for until
+// specified number of controlling objects have all pods running.
+func (w *waitForControlledPodsRunningMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
 	var summaries []measurement.Summary
-	kind, err := util.GetString(config.Params, "kind")
+	action, err := util.GetString(config.Params, "action")
 	if err != nil {
 		return summaries, err
 	}
-	namespace, err := util.GetStringOrDefault(config.Params, "namespace", metav1.NamespaceAll)
+
+	switch action {
+	case "start":
+		w.kind, err = util.GetString(config.Params, "kind")
+		if err != nil {
+			return summaries, err
+		}
+		w.namespace, err = util.GetStringOrDefault(config.Params, "namespace", metav1.NamespaceAll)
+		if err != nil {
+			return summaries, err
+		}
+		w.labelSelector, err = util.GetStringOrDefault(config.Params, "labelSelector", "")
+		if err != nil {
+			return summaries, err
+		}
+		w.fieldSelector, err = util.GetStringOrDefault(config.Params, "fieldSelector", "")
+		if err != nil {
+			return summaries, err
+		}
+		w.operationTimeout, err = util.GetDurationOrDefault(config.Params, "operationTimeout", defaultOperationTimeout)
+		if err != nil {
+			return summaries, err
+		}
+		return summaries, w.start(config.ClientSet)
+	case "gather":
+		syncTimeout, err := util.GetDurationOrDefault(config.Params, "syncTimeout", defaultWaitForRCPodsTimeout)
+		if err != nil {
+			return summaries, err
+		}
+		return summaries, w.gather(config.ClientSet, syncTimeout)
+	case "stop":
+		w.stop()
+		return summaries, nil
+	default:
+		return summaries, fmt.Errorf("unknown action %v", action)
+	}
+}
+
+func (w *waitForControlledPodsRunningMeasurement) start(c clientset.Interface) error {
+	if w.informer != nil {
+		glog.Infof("%v: wait for controlled pods measurement already running", waitForControlledPodsRunningName)
```

Comment:
printing waitForControlledPodsRunningName (which btw is a pointer) doesn't seem to be extremely helpful, I think.

Maybe just print kind, namespace, field and label selector?

[Same below]

Reply:
Or actually, let's introduce `string()` method on a measurement and just use this in every place in logs.

Reply:
> Maybe just print kind, namespace, field and label selector

I think that we discussed this offline (in context of wait_for_pods.go) and decided to change measurement so it uses identifier in logs.

> Or actually, let's introduce string() method on a measurement and just use this in every place in logs.

Let's discuss this offline.

Reply:
After offline discussion, we agreed to add `String()` method to waitForControlledPods measurement. Additionally #275 issue has been created to update all metrics.